GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -768,6 +792,31 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)

```

Comment:
I would rename this to make its intent more clear. It's not meant to remove a path from ModifiedPaths. It's meant to remove it if and only if it's a newly created item, i.e. a temp file. Maybe call it `TryRemoveTempFileFromModifiedPaths`?

Reply:
Maybe not. Now that I read its implementation, I agree with William that it's just doing too much.